chore: update change log to v2.10.0 and install docs for release channels (#12863)
coder/coder: docs/changelogs/v2.10.0.md

## Changelog

> [!NOTE]

> > This is a Mainline Coder release. We advise enterprise customers without a staging environment to install our [latest stable release](https://github.com/coder/coder/releases/latest) while we refine this version. Learn more about how we release in our [docs](../install/releases.md).

Reviewer: ```suggestion
> This is a Mainline Coder release. We advise enterprise customers without a staging environment to install our [latest stable release](https://github.com/coder/coder/releases/latest) while we refine this version. Learn more about our [release schedule](../install/releases.md).
```

## Changelog

> [!NOTE]

> > This is a Mainline Coder release. We advise enterprise customers without a staging environment to install our [latest stable release](https://github.com/coder/coder/releases/latest) while we refine this version. Learn more about how we release in our [docs](../install/releases.md).

Reviewer: Do we want to capitalize mainline?  My preference is lowercase 
<img width="866" alt="Screenshot 2024-04-03 at 3 05 20 PM" src="https://github.com/coder/coder/assets/22407953/3627e1d1-8c91-427a-95ba-6a8668e28740">

Author: Fixed, lowercase only.

### BREAKING CHANGES

> - Removed `max_ttl` from templates (#12644) (@Emyrk)

Reviewer: Can you expand this a bit, my thoughts are the following:

- Explain what `max_ttl` is rendered as in the UI, as this is the API term
- Explain the impact to existing templates who have this applied
- Explain what feature(s) people can use instead of max_ttl to achieve a similar outcome
- Explain our rationale for removing (e.g. it has been deprecated since x)

Reviewer: > Explain the impact to existing templates who have this applied

Plus for existing workspaces

### Features

- Make agent stats' cardinality configurable (#12535) (@dannykopping)
- Upgrade tailscale fork to set TCP options for performance (#12574) (@spikecurtis)
- Add AWS IAM RDS Database auth driver (#12566) (@f0ssel)
- Support Windows containers in bootstrap script (#12662) (@kylecarbs)
- Add `workspace_id` to `workspace_build` audit logs (#12718) (@sreya)
- Make OAuth2 provider not enterprise-only (#12732) (@code-asher)
- Allow number options with monotonic validation (#12726) (@dannykopping)
- Expose workspace statuses (with details) as a prometheus metric (#12762) (@dannykopping)

> - Support adjusting child proc OOM scores (#12655) (@sreya)

Reviewer: Can you explain what the benefit is to the user and why they may want to turn this on, or if it is purely automatic

Reviewer: (the OOM score one)

Author: I honestly don't know what this does.

- Expose HTTP debug server over tailnet API (#12582) (@johnstcn)
- Show queue position during workspace builds (#12606) (@dannykopping)

> - Unhide support bundle cmd (#12745) (@johnstcn)

Reviewer: I think this is screenshot-worthy and worth a slightly larger explanation. I don't think the docs for this are ready yet sadly (#12774) but we can still explain why we built it

## Changelog

> [!NOTE]
> This is a mainline Coder release. We advise enterprise customers without a staging environment to install our [latest stable release](https://github.com/coder/coder/releases/latest) while we refine this version. Learn more about our [Release Schedule](../install/releases.md).

### BREAKING CHANGES

- Removed `max_ttl` from templates (#12644) (@Emyrk)
  > Maximum Workspace Lifetime, or `MAX_TTL`, has been removed from the product in favor of Autostop Requirement. Max Lifetime was designed to automate workspace shutdowns to enable security policy enforcement, enforce routine updates, and reduce idle resource costs.
  >

>   > If you use Maximum Lifetime in your templates, they will no longer be required to shutdown at the end of this timer. Instead, we advise migrating to Autostop Requirement.

Reviewer: ```suggestion
  > If you use Maximum Lifetime in your templates, workspaces will no longer stop at the end of this timer. Instead, we advise migrating to Autostop Requirement.
```
